tests/integrationtest/t/ddl/db_partition.test

```diff
@@ -1,17 +1,14 @@
 # TestDisableTablePartition
```

Comment:
Just remove this test cases

Reply:
I will create another PR for removing tidb_enable_table_partition, since it makes no sense to still have it... And I will remove the test in that PR instead.

Reply:
https://github.com/pingcap/tidb/pull/56190